### PR TITLE
feat(connector-node): docker build connector node using local source code

### PR DESCRIPTION
**Pull request by wenym1**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

As titled. Previously we build the docker image from the source clone from the original connector-node repo. In this PR, we change to build connector node from the local connector source code.

## Checklist For Contributors

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [x] I have demonstrated that backward compatibility is not broken by breaking changes and created issues to track deprecated features to be removed in the future. (Please refer to the issue)
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Checklist For Reviewers

- [x] I have requested macro/micro-benchmarks as this PR can affect performance substantially, and the results are shown.

## Documentation

- [x] My PR **DOES NOT** contain user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -33,14 +33,14 @@ RUN rustup self update \
   && rustup component add rustfmt
 
 RUN cargo fetch
-RUN mkdir -p /risingwave/bin/connector-node
```
Comment (huangjw806) [inline docker/Dockerfile:36]: should be `RUN mkdir -p /risingwave/bin`

Comment (huangjw806) [inline docker/Dockerfile:36]: need keep here

File: docker/Dockerfile [hunk 2]
```diff
@@ -33,14 +33,14 @@ RUN rustup self update \
...
 
 RUN cargo build -p risingwave_cmd_all --release --features "static-link static-log-level" && \
   mv /risingwave/target/release/risingwave /risingwave/bin/ && \
   cargo clean
 
-RUN cd risingwave-connector-node && mvn -B package -Dmaven.test.skip=true
-RUN tar -zxvf /risingwave/risingwave-connector-node/assembly/target/risingwave-connector-1.0.0.tar.gz -C /risingwave/bin/connector-node
+RUN cd /risingwave/java && mvn -B package -Dmaven.test.skip=true && \
+    mkdir -p /risingwave/bin/connector-node && \
```
Comment (huangjw806) [inline docker/Dockerfile:42]: maybe should delete here.

**Review by tabVersion — APPROVED** [review]
ditto

rest LGTM

File: docker/Dockerfile (hunk 1 above)
Comment (wenym1) [reply docker/Dockerfile:36]: fixed.

File: docker/Dockerfile (hunk 2 above)
Comment (wenym1) [reply docker/Dockerfile:42]: This is in the builder, so I think the tar file will not be included in the final image

**Comment by wenym1** [comment]
I have tried triggering a pipeline to build the docker image. I works fine.

https://buildkite.com/risingwavelabs/docker/builds/13440#0186c0f6-bbe5-446d-9092-310c439d1c91

**Pull request merged**